cypress.config.ts

```diff
@@ -1,12 +1,55 @@
 import { defineConfig } from "cypress";
+import * as webpack from "@cypress/webpack-preprocessor";
+import { addCucumberPreprocessorPlugin } from "@badeball/cypress-cucumber-preprocessor";
+
+async function setupNodeEvents(
+  on: Cypress.PluginEvents,
+  config: Cypress.PluginConfigOptions
+): Promise<Cypress.PluginConfigOptions> {
+  // This is required for the preprocessor to be able to generate JSON reports after each run, and more,
+  await addCucumberPreprocessorPlugin(on, config);
+
+  on(
+    "file:preprocessor",
+    webpack({
+      webpackOptions: {
+        resolve: {
+          extensions: [".ts", ".js"],
+        },
+        module: {
+          rules: [
+            {
+              test: /\.ts$/,
+              exclude: [/node_modules/],
+              use: [
+                {
+                  loader: "ts-loader",
+                },
+              ],
+            },
+            {
+              test: /\.feature$/,
+              use: [
+                {
+                  loader: "@badeball/cypress-cucumber-preprocessor/webpack",
+                  options: config,
+                },
+              ],
+            },
+          ],
+        },
+      },
+    })
+  );
+
+  // Make sure to return the config object as it might have been modified by the plugin.
+  return config;
+}
 
 export default defineConfig({
   e2e: {
-    setupNodeEvents(on, config) {
-      // implement node event listeners here
-    },
-    // baseUrl: "https://www.gsmarena.com/",
-    viewportHeight: 800,
-    viewportWidth: 1440,
+    baseUrl: "https://duckduckgo.com",
+    specPattern: "**/*.feature",
```

Comment:
You are configuring only .feature files but you also have tests that don't use cucumber, please choose if you want to use cucumber or not

Reply:
i started looking into cucumber just at the end of day, that's why, i will migrate existing tests into cucumber tests

Reply:
Ok, do that and then assign it to me for review